modeling/dataset/samplers/next_item_prediction.py

```diff
@@ -105,3 +107,238 @@ def create_from_config(cls, config, **kwargs):
             num_users=kwargs['num_users'],
             num_items=kwargs['num_items']
         )
+
+
+class MultiDomainNextItemPredictionTrainSampler(MultiDomainTrainSampler, config_name='multi_domain_next_item_prediction'):
+
+    def __init__(self, dataset, num_users, num_items, target_domain, other_domains, negative_samplers):
+        super().__init__()
+        self._dataset = dataset
+        self._num_users = num_users
+        self._num_items = num_items
+        self._negative_samplers = negative_samplers
+        self._target_domain = target_domain
+        self._other_domains = other_domains
+
+        self._user_id_to_idx_cross_domain_mapping = {domain:{} for domain in self._other_domains}
+        for domain in self._other_domains:
+            for idx, el in enumerate(self._dataset[domain]):
+                u_id = el['user.ids'][0]
+                self._user_id_to_idx_cross_domain_mapping[domain][u_id] = idx
+
+    @classmethod
+    def create_from_config(cls, config, **kwargs):
+        negative_samplers = {}
```

Comment:
Optional. It can be re-written in the following way:

```
domains = [config['target_domain']] + ast.literal_eval(config['other_domains'])
negative_samples = {
    BaseNegativeSampler.create_from_config(
        {'type': config['negative_sampler_type']},
         **kwargs
    )
}
```